torchrl/envs/transforms/transforms.py

```diff
@@ -940,6 +940,85 @@ def _pixel_observation(self, spec: TensorSpec) -> None:
         return spec
 
 
+class TargetReturn(Transform):
+    """Sets a target return for the agent to achieve in the environment.
+
+    Target return can be used in goal-conditioned RL algorithms to set a target that should be achieved at the end of an episode by the agent like Upside-Down RL or Decision Transformer.
+
+    Args:
+        target_return (float): target return to be achieved by the agent.
+        mode (str): mode to be used to update the target return. Can be either "reduce" or "constant". Default: "reduce".
+
+    """
+
+    def __init__(
+        self,
+        target_return: float,
+        mode: str = "reduce",
+    ):
+        super().__init__(in_keys=["reward"], out_keys=["target_return"])
+        self.in_key = "reward"
+        self.out_key = "target_return"
+        self.target_return = target_return
+        self.mode = mode
+
+    def reset(self, tensordict: TensorDict):
+        init_target_return = torch.full(
+            size=(*tensordict.batch_size, 1),
+            fill_value=self.target_return,
+            dtype=torch.float32,
+            device=tensordict.device,
+        )
+        tensordict.set(self.out_key, init_target_return)
+        return tensordict
+
+    def _call(self, tensordict: TensorDict) -> TensorDict:
+        new_target_return = self._apply_transform(
+            tensordict[self.in_key], tensordict[self.out_key]
```

Comment:
tensordict indexing should be achieved via `get` and not `__getitem__` (for efficiency and compatibility with tensorclass)